core/src/main/java/tc/oc/pgm/stats/StatsMatchModule.java

```diff
@@ -238,33 +236,21 @@ public void onFlagDrop(FlagStateChangeEvent event) {
   @EventHandler(priority = EventPriority.MONITOR)
   public void onPlayerDeath(MatchPlayerDeathEvent event) {
     MatchPlayer victim = event.getVictim();
-    MatchPlayer murderer = null;
-
-    if (event.getKiller() != null)
-      murderer = event.getKiller().getParty().getPlayer(event.getKiller().getId());
-
-    PlayerStats victimStats = getPlayerStat(victim);
-
-    victimStats.onDeath();
-
-    sendPlayerStats(victim, victimStats);
-
-    if (murderer != null
-        && PlayerRelation.get(victim.getParticipantState(), murderer) != PlayerRelation.ALLY
-        && PlayerRelation.get(victim.getParticipantState(), murderer) != PlayerRelation.SELF) {
-
-      PlayerStats murdererStats = getPlayerStat(murderer);
-
-      if (event.getDamageInfo() instanceof ProjectileInfo) {
-        murdererStats.setLongestBowKill(victim
-            .getState()
-            .getLocation()
-            .distance(((ProjectileInfo) event.getDamageInfo()).getOrigin()));
-      }
-
-      murdererStats.onMurder();
+    getPlayerStat(victim).onDeath(victim);
+
+    if (event.isChallengeKill()) {
+      MatchPlayerState killer = event.getKiller();
+      assert killer != null;
```

Comment:
Why do you have an assert here?

Reply:
mostly to keep the IDE quiet, if it's a challenge kill we know for certain it's non-null. null killer would be a NONE relationship which doesn't qualify as ENEMY, which is needed for it to be a challenge kill